sys/tools/syscalls/config.lua

```diff
@@ -0,0 +1,321 @@
+					    nextline)
+				end
+			end
+			cfg[key] = value
+		elseif not nextline:match("^%s*$") then
+			-- Make sure format violations don't get overlooked
+			-- here, but ignore blank lines.  Comments are already
+			-- stripped above.
+			abort(1, "Malformed config line: " .. nextline)
+		end
+	end
+
+	assert(io.close(fh))
+	return cfg
+end
+
+-- Merges processed configuration file into the global config map (see above),
+-- or returns NIL and a message.
+function config.merge(fh)
+    if fh ~= nil then
+    	local res = assert(config.process(fh))
+    
+    	for k, v in pairs(res) do
+    		if v ~= config[k] then
+                -- handling of sets
+                if v:find("abi_flags") then
+                    -- match for pipe, that's how abi_flags is formatted
+                    table.insert(config[k], util.setFromString(v, "[^|]+"))
+                elseif v:find("capenabled") or
+                       v:find("syscall_abi_change") or
+                       v:find("syscall_no_abi_change") or
+                       v:find("obsol") or
+                       v:find("unimpl") then
+                    -- match for space, that's how these are formatted
+                    table.insert(config[k], util.setFromString(v, "[^ ]+"))
+                else
+    			    config[k] = v
+                end
+                -- construct config modified table as config is processed
+                config.mod[k] = true
+    		end
+            config.mod[k] = false  -- config wasn't modified
+    	end
+    end
+end
+
+-- Returns TRUE if there are ABI changes from native for the provided ABI flag. 
+function config.abiChanges(name)
+	if config.known_abi_flags[name] == nil then
+		util.abort(1, "abi_changes: unknown flag: " .. name)
+	end
+    return config.abi_flags[name] ~= nil
+end
+
+-- Merge any changes to the ABI (from native) and handles if there shouldn't be
+-- changes.
+function config.mergeChangesAbi()
+    if config.no_changes_abi then 
+        config.changes_abi = false
+    end
+end
+
+--
+-- Call to instantiate config.compat_set with configuration file compatability
+-- options (may remain empty if no compatability options are required (e.g., 
+-- native)).
+-- 
+function config.mergeCompat()
+    if config.compat_set ~= "" then
+    	if not compat_option_sets[config.compat_set] then
+    		util.abort(1, "Undefined compat set: " .. compat_set)
+    	end
+    
+    	config.compat_options = compat_option_sets[config.compat_set]
+    else
+    	config.compat_options = {}
+    end
+end
+
+--
+-- Parses the provided capabilities.conf, returns a string to be split and 
+-- merged into the global config.
+-- Helper function for config.capability() to use.
+--
+-- NOTE: Hasn't been changed from makesyscalls.lua, will work the same.
+--
+local function grabCapenabled(file, open_fail_ok)
+	local capentries = {}
+	local commentExpr = "#.*"
+
+	if file == nil then
+		print "No file"
+		return {}
+	end
+
+	local fh = io.open(file)
```

Comment:
`io.open` returns nil, msg, errno in the case of failure.  We should at least preserve the msg so that we can pass it on in the `abort()` below in the `not open_fail_ok` case